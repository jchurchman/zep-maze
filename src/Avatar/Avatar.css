:local(.avatar) {
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  padding: auto;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 10000;
}
