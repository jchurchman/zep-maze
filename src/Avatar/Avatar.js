import React from 'react';
import styled from 'styled-components';

const Sprite = styled.span`
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    padding: auto;
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 1000;
    font-size: 15vw;
`

const transform = `translate(0.6vw, -0.4vw)`;

export default function Avatar() {
    
    return ( 
        <Sprite>
            <span style={{transform}} role="img" aria-label="neutral face">😐</span>
        </Sprite>
    );
}
