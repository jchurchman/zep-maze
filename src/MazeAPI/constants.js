export const FETCHING_MAZES = 'FETCHING_MAZES';
export const FETCHING_MAZE = 'FETCHING_MAZE';
export const FETCHED_MAZES = 'FETCHED_MAZES';
export const FETCHED_MAZE = 'FETCHED_MAZE';
export const FETCHED_MAZE_ERROR = 'FETCHED_MAZE_ERROR';
export const ADDING_MAZE = 'ADDING_MAZE';
export const ADDED_MAZE = 'ADDED_MAZE';
export const ADDING_MAZE_ERROR = 'ADDING_MAZE_ERROR';