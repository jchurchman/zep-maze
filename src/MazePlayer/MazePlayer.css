:local(.player) {
  margin: 20px auto; }
  :local(.player) .viewport {
    width: 146px;
    height: 146px;
    overflow: hidden;
    position: relative;
    margin: auto;
    border: 2px solid orange; }
