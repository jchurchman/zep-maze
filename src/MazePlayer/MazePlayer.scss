$size: 146px;


:local(.player){
    margin: 20px auto;
    .viewport{
        width: $size;
        height: $size;
        overflow: hidden;
        position: relative;
        margin: auto;
        border: 2px solid orange;
    }
}