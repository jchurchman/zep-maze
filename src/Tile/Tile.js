import React from 'react';
import styled from 'styled-components';

const Square = styled.td`
    height: 16vw;
    width: 16vw;
    box-sizing: border-box;
    border: 6px outset #a9a9a9;
    text-align: center;
    vertical-align: middle;
`

export function Tile({ value = '' }) {
    return (
        <Square className={`${value.toLowerCase()}`}>
        </Square>
    );
}