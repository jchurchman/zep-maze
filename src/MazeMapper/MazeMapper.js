import React, { Component } from 'react';
import { Tile } from '../Tile/Tile';
import styles from './MazeMapper.css';

export default class MazeMapper extends Component{

    render() {
        const { position, maze } = this.props;


        const xVal = (position[0] - 1) * -48;
        const yVal = (position[1] - 1) * -48;
        
        const transform = `translate( ${ xVal }px, ${ yVal }px )`;

        return (
            <table className={styles.maze} style={{transform}}>
                <tbody>
                    {maze.matrix.map( ( row, index ) => {
                        return (
                            <tr key={index}>
                            {row.map( ( tile, index ) => {
                                return <Tile key={index} value={tile}/>
                            })}
                            </tr>
                        )
                    })}
                </tbody>
            </table>
    )}
}