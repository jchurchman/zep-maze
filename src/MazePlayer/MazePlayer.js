import React from 'react';
import MazeMapper from '../MazeMapper/MazeMapper';
import Avatar from '../Avatar/Avatar';
import ModalContainer from '../Modal/ModalContainer';
import ControllerContainer from '../Controller/ControllerContainer';
import styled from 'styled-components';

const Player = styled.div`
    margin: 4vw;
    display: flex;
    justify-content: space-around;
    align-items: center;
`

const Viewport = styled.div`
    width: 48vw;
    height: 48vw;
    overflow: hidden;
    position: relative;
    margin: auto;
    border: 6px outset #D0AC4C;
`

export default function MazePlayer({ game, status }) {
    const { maze, error, loading } = game;
    const { position } = status;

    if( loading || !maze ) return <div>Loading Mazes...</div>;
    if (error) return <div>There was an error loading this maze, try refreshing the page</div>;

    return (

        <Player>
            <Viewport>
                <Avatar />
                <MazeMapper maze={maze} position={position}/>
            </Viewport>
            <ModalContainer>
                <p>Congratulations, you solved the maze!</p>
            </ModalContainer>
            <ControllerContainer />
        </Player>
    )
}