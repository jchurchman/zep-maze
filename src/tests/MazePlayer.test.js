import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MazePlayer from '../MazePlayer/MazePlayer';
import Avatar from '../Avatar/Avatar';
import MazeMapper from '../MazeMapper/MazeMapper';
import { WALL, FLOOR, START, END } from '../Tile/Tile.constants';

const maze = {
    score: [],
    matrix: [
        [ WALL, WALL, WALL, WALL, WALL ],
        [ WALL, START, WALL, END, WALL ],
        [ WALL, FLOOR, WALL, FLOOR, WALL ],
        [ WALL, FLOOR, FLOOR, FLOOR, WALL ],
        [ WALL, WALL, WALL, WALL, WALL ],
    ]
}

const badGame = {
    maze: null, 
    loading: false,
    error: null
}

const goodGame = {
    maze: maze,
    loading: false,
    error: null
};

const defaultStatus = {
    position: [ 1, 1 ],
    activity: 'PLAYING'
}

describe('MazePlayer component', () => {
    it('renders a  loading div when no maze passed', () => {
        const wrapper = shallow(<MazePlayer 
            game={badGame}
            status={defaultStatus}
        />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders a div with a maze and an avatar when child components passed', () => {
        const wrapper = shallow(
            <MazePlayer 
                game={goodGame}
                status={defaultStatus}
            />
            );
        expect(toJSON(wrapper)).toMatchSnapshot();
    })
})