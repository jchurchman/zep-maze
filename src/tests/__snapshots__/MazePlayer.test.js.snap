// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MazePlayer component renders a  loading div when no maze passed 1`] = `
<div>
  Loading Mazes...
</div>
`;

exports[`MazePlayer component renders a div with a maze and an avatar when child components passed 1`] = `
<styled.div>
  <styled.div>
    <Avatar />
    <MazeMapper
      maze={
        Object {
          "matrix": Array [
            Array [
              "WALL",
              "WALL",
              "WALL",
              "WALL",
              "WALL",
            ],
            Array [
              "WALL",
              "START",
              "WALL",
              "END",
              "WALL",
            ],
            Array [
              "WALL",
              "FLOOR",
              "WALL",
              "FLOOR",
              "WALL",
            ],
            Array [
              "WALL",
              "FLOOR",
              "FLOOR",
              "FLOOR",
              "WALL",
            ],
            Array [
              "WALL",
              "WALL",
              "WALL",
              "WALL",
              "WALL",
            ],
          ],
          "score": Array [],
        }
      }
      position={
        Array [
          1,
          1,
        ]
      }
    />
  </styled.div>
  <Connect(Modal)>
    <p>
      Congratulations, you solved the maze!
    </p>
  </Connect(Modal)>
  <Connect(Controller) />
</styled.div>
`;
